platforms/core-execution/request-handler-worker/build.gradle.kts

```diff
@@ -13,23 +13,25 @@ dependencies {
     api(projects.baseServices)
     api(projects.buildOperations)
     api(projects.concurrent)
+    api(projects.hashing)
+    // TODO: Ideally, we would not depend on model-core in a worker.
+    // All we really want is the instantiation infrastructure, but this
+    // brings in core-api, which should be avoided in workers.
+    api(projects.modelCore)
     api(projects.messaging)
+    api(projects.problemsApi)
     api(projects.serialization)
+    api(projects.snapshotsWorker)
+    api(projects.stdlibJavaExtensions)
     api(projects.workerMain)
 
     implementation(projects.classloaders)
+    implementation(projects.coreApi)
```

Comment:
We should group this `core-api` dependency along with the existing TODO, or add a similar TODO here.